examples/file-hash/src/codec.rs

```diff
@@ -0,0 +1,88 @@
+use std::cell::RefCell;
+
+use gloo_worker::Codec;
+use js_sys::{Array, JsString, Reflect};
+use wasm_bindgen::JsCast;
+use web_sys::File;
+
+use crate::HashInput;
+
+thread_local! {
+    static FILE_STORE: RefCell<Option<File>> = RefCell::new(None);
+}
+
+pub struct TransferrableCodec {}
+
+impl TransferrableCodec {
+    pub fn pre_encode_input(input: &HashInput) {
+        let file = input.file.clone();
+        FILE_STORE.with(|m| m.replace(file));
+    }
+
+    pub fn post_decode_input(input: &mut HashInput) {
+        let f = FILE_STORE.with(|m| m.take());
+        input.file = f;
+    }
+}
+
+// This codec implementation relys on some internal implementation details about gloo worker message types.
+// This should be considered as a last resort approach after all other options are exhausted.
+//
+// In addition, using this approach could also potentially create unsound code.
+// This could cause data being sent to a wrong worker or with wrong handler id if it is not implemented
+// properly.
+//
+// You should only use this approach if you know how to make it sound.
+// This example mitigates this issue by only allowing 1 file to be processed at a time
+// and only 1 worker instance across the entire tab.
```

Comment:
This sounds like a pretty big gotcha, especially for an example. It _can_ work but I would argue this is broken.

The usecase I had for this required me to fork a worker from a svelte component and pass callback to the fork that mutates the component's state. I certainly don't see that working with this implementation

Reply:
I agree with you.

However, I do not have a better way of achieving it and I do not like the way I made it to work either.

The proper solution would be to wait until https://github.com/cloudflare/serde-wasm-bindgen/issues/32 resolved (there is a pending pull request), which brings `Serialize` and `Deserialize` to `JsValue`. The codec will work with transferrable types by simply using a codec implemented with `serde-wasm-bindgen` after that issue is resolved.